Dockerfile

```diff
@@ -0,0 +1,78 @@
+ARG GOLANG_VERSION
+FROM library/golang:${GOLANG_VERSION}-alpine AS build
```

Comment:
This will download the alpine image of the architecture of the build machine, not the target machine.
And in flannel-io: https://github.com/flannel-io/flannel/blob/master/Documentation/kube-flannel.yml#L170-L175
That 'cp' command will be run on the target architecture.
On an 'arm' machine with this image build on an amd64 machine, 'cp' will fail with exec format error.

Have I understood it wrong?

Reply:
Hi Rajat - 

The dockerfile is strictly for building the binaries in a reproducible manner for dev/debug/qa purposes. This PR will remove all image builds for cni-plugin and instead move forward with publishing binaries and tar.gz for each arch.

Reply:
An update on this - I have added back all of the image build and publish steps after discussion with @Oats87, @manuelbuil , and @luthermonson . A potential future goal is to remove the need to build and publish the cni-plugin images by consuming the binaries that we will now be publishing automatically in the flannel image itself.

Reply:
per conversation on meetup @rajatchopra is good with his comment since @rosskirkpat added back the images